pkg/net/ip.go

```diff
@@ -242,3 +244,62 @@ func IPString(ip cue.Value) (string, error) {
 	}
 	return ipdata.String(), nil
 }
+
+func netIPAddOffset(addr netip.Addr, offset *big.Int) (netip.Addr, error) {
+	i := big.NewInt(0).SetBytes(addr.AsSlice())
+	i = i.Add(i, offset)
+
+	if i.Sign() < 0 {
+		return netip.Addr{}, errors.New("underflowed")
+	}
+
+	b := i.Bytes()
+	size := addr.BitLen() / 8
+
+	if len(b) > size {
+		return netip.Addr{}, errors.New("overflowed")
+	}
+
+	if len(b) < size {
+		b = append(make([]byte, size-len(b)), b...)
```

Comment:
It's easy to avoid an extra allocation here by making the slice the correct size up front:

```suggestion
		b = append(make([]byte, size-len(b), size), b...)
```